### PR TITLE
add normalizations for FFT functions

### LINKED ISSUE #8100: Enhancement: support normalization options in FFT
numpy 1.20 and later supports [several alternate normalizations](https://numpy.org/doc/stable/reference/routines.fft.html#normalization) for FFT functions, including `"backward"`, `"ortho"`, and `"forward"`. We should support these in `jax.numpy.fft` as well.

### PR DESCRIPTION
**Pull request by jcopo**
add several alternate normalizations for fft and ifft functions, including "backward", "ortho", and "forward" in jax.numpy.fft. #8100

### CODE REVIEW

**Review by jakevdp — CHANGES_REQUESTED** [review]
Looks great! Thanks for working on this.

A couple comments below... also, once you're finished please plan to squash all changes into a single commit (see https://jax.readthedocs.io/en/latest/contributing.html#single-change-commits-and-pull-requests)

Thanks!

**Review by jakevdp — APPROVED** [review]
Looks great, thanks!

**Review by jakevdp — APPROVED** [review]
Nice! Let's see if this passes the tests now.

**Review by jakevdp — APPROVED** [review]
